extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/schema/SchemaManagementIntegrator.java

```diff
@@ -47,8 +47,11 @@ public void integrate(Metadata metadata, SessionFactoryImplementor sessionFactor
 
     @Override
     public void disintegrate(SessionFactoryImplementor sessionFactory, SessionFactoryServiceRegistry serviceRegistry) {
-        metadataMap.remove(nameCache.get(sessionFactory));
-        nameCache.remove(sessionFactory);
+        final String name = nameCache.get(sessionFactory);
+        if (name != null) {
+            metadataMap.remove(name);
+            nameCache.remove(sessionFactory);
+        }
```

Comment:
Isn't better to write like: 

```suggestion
        final String name = nameCache.remove(sessionFactory);
        if (name != null) {
            metadataMap.remove(name);
        }
```

Reply:
Good point :)